core/templates/components/state-editor/state-interaction-editor/state-interaction-editor.component.ts

```diff
@@ -191,13 +189,14 @@ export class StateInteractionEditorComponent
         this.updateDefaultTerminalStateContentIfEmpty();
       }
       this.stateInteractionIdService.saveDisplayedValue();
-      this.onSaveInteractionId.emit(this.stateInteractionIdService.displayed);
     }
-
     this.stateCustomizationArgsService.saveDisplayedValue();
-    this.onSaveInteractionCustomizationArgs.emit(
-      this.stateCustomizationArgsService.displayed
-    );
+
+    this.interactionData.interactionId =
+    this.stateInteractionIdService.displayed;
+    this.interactionData.customizationArgs =
+    this.stateCustomizationArgsService.displayed;
```

Comment:
Why does interactionData need to be defined on `this`? Try to avoid storing state where not needed.

Also the indentation here is not right, if you are continuing a line you need to use indentation to add a scope. But it would be better if you just created an interactionData object here directly, I think, rather than modifying an existing one.

```suggestion
    this.interactionData.interactionId = (
      this.stateInteractionIdService.displayed);
    this.interactionData.customizationArgs = (
      this.stateCustomizationArgsService.displayed);
```

Reply:
Done